### ISSUE TITLE
Refactor how we handle AblyAuth.ClientId

### ISSUE THREAD

**Issue by marto83** [ably/ably-dotnet#930]
Currently the ClientId just delegates to a number of other internal properties like `ConnectionClientId`, `CurrentToken.ClientId`, `CurrentTokenParams.ClientId`. Which was all good until we needed to know when the `ClientId` has changed. It will be better to update `AblyAuth.ClientId` to be a simple property that is changed when required. This way it will be much simpler to implement the notification handler.

┆Issue is synchronized with this [Jira Task](https://ably.atlassian.net/browse/SDK-1258) by [Unito](https://www.unito.io)

**Comment by tomkirbygreen** [comment]
@marto83 Can I close this issue now that https://github.com/ably/ably-dotnet/pull/931 has landed on `main`?

**Comment by sacOO7** [comment]
Yeah, I think we should close this issue. @marto83 you can check again if the issue is resolved.
